test/iterators_test.jl

```diff
@@ -0,0 +1,69 @@
+@testitem "iterators.jl" begin 
+    @testset "SubsetGrayCode iterates over all weight k subsets of {1,..,n}" begin
+        len = 15
+        weight = 7
+        sgc = CodingTheory.SubsetGrayCode(len, weight) 
```

Comment:
to keep ```SubsetGrayCode``` simpler I did not add a mutate keyword. I could not see a need for it because if we need a collection of all subsets we can use the subsets function in ```Hecke```